o2a/converter/workflow.py

```diff
@@ -24,21 +24,14 @@
 class Workflow:  # pylint: disable=too-few-public-methods
     """Class for Workflow"""
 
-    dag_name: Optional[str]
-    input_directory_path: str
-    output_directory_path: str
-    relations: Set[Relation]
-    nodes: Dict[str, ParsedActionNode]
-    dependencies: Set[str]
-
     def __init__(
         self,
-        input_directory_path,
-        output_directory_path,
-        dag_name=None,
-        relations=None,
-        nodes=None,
-        dependencies=None,
+        input_directory_path: str,
```

Comment:
We agreed that a better idea is to define types when declaring variables, not when constructor parameters are declared.

Reply:
Yeajm but since then I checked that intellij does not infer types correctly in this case. So it works better for the tool if types are defined in constructor arguments where possible because then inference works !